### PR TITLE
Added AllotropyException to be used instead of Generic Exception

### PR DESCRIPTION
**Pull request by EihabSyed**

### CODE REVIEW

File: src/allotropy/allotrope/allotrope.py [hunk 1]
```diff
@@ -19,6 +19,10 @@ class AllotropeConversionError(Exception):
     pass
 
 
+class AllotropeException(Exception):
```
Comment (mdlu) [inline src/allotropy/allotrope/allotrope.py:22]: could you rename this `AllotropyException`?

**Review by mdlu — CHANGES_REQUESTED** [review]
this is a great start, could you also replace all instances of `Exception` in the repo, not just in the Envision parser?

**Review by slopez-b — COMMENTED** [review]
Any reason to create a new exception instead of using the existing `AllotropeConversionError`?

**Comment by EihabSyed** [comment]
> Any reason to create a new exception instead of using the existing `AllotropeConversionError`?

The `AllotropeConversionError` was a specific error for conversion whereas the new `AllotropyException` is meant to provide more info as to where the error originates from, in this case the the error is from `Allotropy`

**Review by mdlu — DISMISSED** [review]

**Review by mdlu — APPROVED** [review]

**Pull request merged**